hack/init_cloud_storage_source.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: ./init_cloud_storage_source.sh
+# The current project set in gcloud MUST be the same as where the cluster is running.
+source $(dirname $0)/lib.sh
+
+PUBSUB_SERVICE_ACCOUNT_KEY_TEMP="cre-pubsub.json"
+
+# Download a JSON key for the service account.
+gcloud iam service-accounts keys create ${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP} \
```

Comment:
Does this fail if the file already exists?

Reply:
Yes. It will fail. But every time when run the script, we will delete the file in the end with `
rm ${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}`. Therefore, we can run this script multiple times.  Unless user has one file with the same name created by himself,  this failure won't happen. Or do you think we should allow user to pass the  `PUBSUB_SERVICE_ACCOUNT_KEY_TEMP` that they want to use?

Reply:
It feels like it might be nicer to make the key name a little more obscure (so we don't have to worry about unrelated keys matching it), but also unilaterally delete it before this line.  I'm not 100% fond of deleting things i'm not sure we created, but as is if a previous run exits this script without finishing it (ctrl-c etc) the next run will fail in an obscure way due to the existence of this file.  alternately, may just check if this file exists prior to this call and tell the user to delete it if needed?

Reply:
That said, this isn't a regression as far as i can tell, so i could live with it as-is.

Reply:
Currently the key name matches what the manual guide is using https://github.com/google/knative-gcp/blob/master/docs/install/authentication-mechanisms-gcp.md#option-2-kubernetes-secrets and https://github.com/google/knative-gcp/blob/master/docs/install/pubsub-service-account.md#option-2-export-service-account-keys-and-store-them-as-kubernetes-secrets.  Just use the same name for consistency. I thought we use such key name for simplicity.  Do you think we should use more obscure name in both init_xxx scripts and manual guide?As long as we keep key name consistent, I think it will be fine.

Reply:
I suggest using `mktemp` to create temporary directories for each run. Then there's no state to consider on start or exit.

Reply:
I'm good putting it all in mktemp if that's doable.  Makes it pretty obvious they are disposable directories.

Reply:
`gcloud iam service-accounts keys create ${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP} \` this command won't fail when the file already exists. It will override the file. I will put it all in mktemp for both manual steps and init_xxx scripts.